examples/nlp_seq/train.py

```diff
@@ -191,35 +192,38 @@ def compute_metrics(logits, labels, weights):
   return metrics
 
 
-def train_step(optimizer, batch, learning_rate_fn, model, dropout_rng=None):
+def train_step(state,
+               batch,
+               model,
+               learning_rate_fn,
+               dropout_rng=None):
   """Perform a single training step."""
-
   train_keys = ['inputs', 'targets']
   (inputs, targets) = [batch.get(k, None) for k in train_keys]
 
   weights = jnp.where(targets > 0, 1, 0).astype(jnp.float32)
-  dropout_rng, new_dropout_rng = random.split(dropout_rng)
+
+  dropout_rng = jax.random.fold_in(dropout_rng, state.step)
+
   def loss_fn(params):
-    """Loss function used for training."""
+    """loss function used for training."""
     logits = model.apply({'params': params}, inputs=inputs, train=True,
                          rngs={'dropout': dropout_rng})
     loss, weight_sum = compute_weighted_cross_entropy(logits, targets, weights)
+
     mean_loss = loss / weight_sum
     return mean_loss, logits
 
-  step = optimizer.state.step
+  step = state.step
```

Comment:
nit: is this variable needed? it's only used once below (and once above where you refer to it by `state.step`)

Reply:
No. Changed it.

Reply:
did you push the changes?